src/com/vivian/java/leetcode/Q434_NumberofSegmentsinaString.java

```diff
@@ -11,5 +11,30 @@
  * Input: "Hello, my name is John" Output: 5
  */
 public class Q434_NumberofSegmentsinaString {
-
+    public int numberOfSegmentSinaString(String str) {
+        int count = 0;
+        boolean flag=false;
+        if (str.isEmpty() || str == null) {
+            return 0;
+        }
+        for (int i = 0; i < str.length(); i++) {
+            char c = str.charAt(i);
+            if (c==' ') {
+                if(flag==false){
```

Comment:
if (flag == false) is equivalent to if (!flag), and the latter code is more clear. So please, change to the latter one.